### PR TITLE
implement HeaderField

### PR DESCRIPTION
**Pull request by copepod**
please review

### CODE REVIEW

File: viritin/src/main/java/org/vaadin/viritin/fields/HeaderField.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ * <p>
+ * The creation of the header can be controlled via a
+ * {@link org.vaadin.viritin.fields.CaptionGenerator}.
+ *
+ * @author Birke Heeren
+ *
+ * @param <T> the type of the entity
+ */
+public class HeaderField<T> extends LabelField<T>
+        {
```
Comment (dve) [inline viritin/src/main/java/org/vaadin/viritin/fields/HeaderField.java]: Code format

File: viritin/src/main/java/org/vaadin/viritin/fields/HeaderField.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    
+    private static final long serialVersionUID = -6461526936461789654L;
+
+    public HeaderField(String caption) {
+        super(caption);
+        label = new Header("");
```
Comment (dve) [inline viritin/src/main/java/org/vaadin/viritin/fields/HeaderField.java]: maybe use `setLabel` set label instead

File: viritin/src/main/java/org/vaadin/viritin/fields/HeaderField.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    public HeaderField() {
+	label = new Header("");
+    }   
+
+    public Header getLabel() {
+        return (Header)label;
```
Comment (dve) [inline viritin/src/main/java/org/vaadin/viritin/fields/HeaderField.java]: maybe use `(Header) getLabel` instead?

File: viritin/src/main/java/org/vaadin/viritin/fields/HeaderField.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    public void setHeaderLevel(int headerLevel) {
+	((Header)label).setHeaderLevel(headerLevel);
```
Comment (dve) [inline viritin/src/main/java/org/vaadin/viritin/fields/HeaderField.java]: Avoid casting again with using  `getLabel`

File: viritin/src/main/java/org/vaadin/viritin/fields/LabelField.java [hunk 5]
```diff
@@ -58,7 +58,7 @@ public LabelField(String caption) {
     }
 
     private CaptionGenerator<T> captionGenerator = new ToStringCaptionGenerator<>();
-    private Label label = new RichText();
+    protected Label label = new RichText();
```
Comment (dve) [inline viritin/src/main/java/org/vaadin/viritin/fields/LabelField.java]: If we override `getLabel`in `HeaderField` we don't have to expose the label propertie

**Comment by dve** [comment]
Looks fine to me!
@mstahv  what do you think?

**Comment by mstahv** [comment]
Looks good to me, go ahead and merge it.

**Pull request merged**